Automattic/wp-calypso
# Refactor: Site Settings Export components

The components under `client/my-sites/exporter` had grown over time and the structure was becoming confusing. This arranges the export and guided transfer component files into a heirarchy more closely matching the way they are displayed on the page, and improves separation of concerns.

Prior to this change, the export component and guided transfer state were intermingled with a long list of props. This separates the two and adds Redux connectors where necessary to ensure each card only has access to the state it requires.

### Before

 - exporter
   - index.jsx
   - exporter.jsx
   - advanced-settings.jsx
   - guided-transfer-options.jsx
   - guided-transfer-details.jsx
   - guided-transfer-in-progress.jsx
   - spinner-button.jsx
   - style.scss

### After

 - exporter
   - export-card
     - index.jsx
     - advanced-settings.jsx
     - post-type-options.jsx
     - select.jsx
     - spinner-button.jsx
   - guided-transfer-card
     - index.jsx
     - in-progress.jsx
   - index.jsx
   - notices.jsx
   - style.scss


### How to test

1. Go to **My Sites > Settings > Export**
2. Select custom export content, then click Export. Ensure the export completes as usual and a success notice appears.
3. Click Purchase a Guided Transfer and you should be redirected to the host selection screen (which is not affected by this PR)

Test live: https://calypso.live/?branch=update/exporter/reorganize-components
